pkg/alertmanager/validation.go

```diff
@@ -115,6 +115,10 @@ func validatePagerDutyConfigs(configs []monitoringv1alpha1.PagerDutyConfig) erro
 		if conf.RoutingKey == nil && conf.ServiceKey == nil {
 			return errors.New("one of 'routingKey' or 'serviceKey' is required")
 		}
+
+		if err := conf.HTTPConfig.Validate(); err != nil {
```

Comment:
hmm so we didn't validate the HTTP config for all receivers?

Reply:
we do validate, but down at the "check" level (not here at admission webhook level)- https://github.com/prometheus-operator/prometheus-operator/blob/1090bed455c51b7525f78a9c26e942d839bc0eb5/pkg/alertmanager/operator.go#L1371

my thoughts were that we were reading the secrets etc at that point but looking closer at some of those functions I think we might be able to include them here also

Reply:
ok so i've included what is safe to validate here also. We cant remove the validation from where it already exists since other packages rely on it also. We could potentially rethink the overall "validation strategy" of the project if we wanted to.